src/FiberMap.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace React\Async;
+
+use Fiber;
+use React\Promise\PromiseInterface;
+
+/**
+ * @internal
+ */
+final class FiberMap
+{
+    private array $status = [];
+    private array $map   = [];
+
+    public function register(Fiber $fiber): void
+    {
+        $this->status[spl_object_hash($fiber)] = false;
+        $this->map[spl_object_hash($fiber)] = [];
```

Comment:
Makes me wonder if `\spl_object_id()` might be more efficient/faster here? (possible micro-optimization)

nit: prefix global functions like `\spl_object_hash()` for consistency?